### DISCUSSION TITLE
Unknown Value 6_28_4 - Current Discoveries

### DISCUSSION THREAD

**Discussion by BagonGanda** [Deijin27/RanseiLink#57] [Show and tell]
As of now, I've been looking at the functions of Unknown Value 6_28_4 on the moves page, and seeing what exactly it does. I can't say for sure that this is exactly what they do, but so far, this has lined up.

Unknown Value 6_28_4 is tied directly to Movement Animation in some way. The only Unknown Value 6_28_4 values that are used are 0, 1, 2, 4, and 8. 0 is exclusively used on Fiery Dance only. 1 is exclusive to the unique "movement" moves, being Splash, Teleport, Dig, and Sky Drop. 4 is exclusive to Volt Switch. 2 is for any other moves that have Movement Animation, and 8 is for moves that do not have one.

At first glance, I assumed that using 4 for Unknown Value 6_28_4 made it so that the movement animation didn't look weird for backwards movement, same so for 2 for forwards movement, but this had outliers (as Tackle, Quick Attack, and other moves did not actually move the user a tile, but did have movement animation and the Unknown Value 6_28_4 of 2), so I knew this could not be it. To troubleshoot, I gave a move with the Fiery Dance movement animation and the ability of the pokemon moving two tiles forward the Unknown Value 6_28_4 of 4 to see what is actually happening, and I think I have figured it out a little. I gave the move a starting, projectile, and impact animation to see if this number affects those as well.

Here's my results in as full detail as possible.

Unknown Value 6_28_4 of 0- Pokemon does its movement animation, moving two squares forward via Fiery Dance animation, then does starting, projectile, then impact animation.
Unknown Value 6_28_4 of 1- Pokemon starts its starting animation at the same time its movement animation begins, arrives at its destination, then does projectile and impact animation.
Unknown Value 6_28_4 of 2- Pokemon starts its starting animation, then movement animation begins. A little earlier than halfway through the movement animation, the projectile animation happens. when it reaches its destination, it does the impact animation.
Unknown Value 6_28_4 of 4- Pokemon does its starting animation, projectile animation, then impact animation. then, it conducts its movement animation and arrives at its destination.
Unknown Value 6_28_4 of 8- Pokemon does its starting, projectile, and impact animation, movement animation doesnt happen, Pokemon does not move.

and to test unused variables, i tried Unknown Value 6_28_4's of 3 and 5, and these were the following results.

Unknown Value 6_28_4 of 3- Pokemon does its starting animation and impact animation, then conducts its movement animation. halfway through the movement animation, the impact animation happens, and then the pokemon reaches its destination.
Unknown Value 6_28_4 of 5- same as unknown value of 8.

For now, I'm going to assume that 5 through 15 function the exact same as each other, and they simply chose 8 since it would stick to multiples of 2, or whatever. In a reply in a moment, i will post videos of everything detailed here as well

**Comment by BagonGanda** [comment]
Unknown Value 6_28_4 value of 0-
https://github.com/Deijin27/RanseiLink/assets/84983046/95a2e41a-68c0-4f80-96e0-02badb2d0b1d


Unknown Value 6_28_4 value of 1-
https://github.com/Deijin27/RanseiLink/assets/84983046/c5a0af77-e5af-4bd2-9e96-aaff96e3b0a6

Unknown Value 6_28_4 value of 2-
https://github.com/Deijin27/RanseiLink/assets/84983046/df2a8b4f-972e-425b-a99b-ee66a9fe0216

Unknown Value 6_28_4 value of 3-
https://github.com/Deijin27/RanseiLink/assets/84983046/42e59fa6-56e9-42a3-9e10-f714f2c2a332

Unknown Value 6_28_4 value of 4-
https://github.com/Deijin27/RanseiLink/assets/84983046/e1df3f2e-1955-42cf-ae70-5299a119cae0

Unknown Value 6_28_4 value of 5-
https://github.com/Deijin27/RanseiLink/assets/84983046/73befb47-4506-4bd3-a2b5-110e41f3cf80

Unknown Value 6_28_4 value of 8-
https://github.com/Deijin27/RanseiLink/assets/84983046/6a785ba3-ad1a-4fcf-97fc-ba584df70a4b

**Comment by Deijin27** [comment]
Nice research.

The multiples 2 would lead you to think they're bit flags:
```
0 = 0b0000
1 = 0b0001
2 = 0b0010
4 = 0b0100
8 = 0b1000
```

But then their values being combinable doesn't make sense, nor does 3 seem to be a combination of 2 and 1.
